versions/schematic-2.md

```diff
@@ -0,0 +1,196 @@
+<a name="schematicPaletteMax"></a>PaletteMax | `integer` | Specifies the size of the block palette in number of bytes needed for the maximum palette index. Implementations may use this as a hint for the case that the palette data fits within a datatype smaller than a 32-bit integer that they may allocate a smaller sized array.
+<a name="schematicPalette"></a>Palette | [Palette Object](#paletteObject) | Specifies the block palette. This is a mapping of block states to indices which are local to this schematic. These indices are used to reference the block states from within the [BlockData array](#schematicBlockData). It is recommeneded for maximum data compression that your indices start at zero and skip no values. The maximum index cannot be greater than [`PaletteMax - 1`](#schematicPaletteMax). While not required it is **highly** recommended that you include a palette in order to tightly pack the block ids included in the data array.
+<a name="schematicBlockData"></a>BlockData | `varint[]` | **Required.** Specifies the main storage array which contains `Width * Height * Length` entries. Each entry is specified as a varint and refers to an index within the [Palette](#schematicPalette). The entries are indexed by `x + z * Width + y * Width * Length`.
+<a name="schematicTileEntities"></a>TileEntities | [TileEntity Object](#tileEntityObject)[] | Specifies additional data for blocks which require extra data. If no additional data is provided for a block which normally requires extra data then it is assumed that the TileEntity for the block is initialized to its default state.
+<a name="schematicEntities"></a>Entities | [Entity Object](#entityObject)[] | Specifies entities to be placed in the schematic. If no additional data is provided for an [entity type](#defEntityType) which normally requires extra data, then it is assumed that the Entity is initialized with all defaults.
+<a name="schematicBiomePaletteMax"></a>BiomePaletteMax | `integer` | Specifies the size of the biome palette s
+<a name="schematicBiomePalette"></a>BiomePalette | [Palette Object](#paletteObject) | Specifies the biome palette. This is a mapping of [biomes](#defBiome) to indicies which are local to this schematic. These indices are used to reference the biomes from within the [Biome array](#schematicBiomeData). It is recommended for maximum data compression that your indices start at zero and skip no values. The maximum index cannot be greater than [`BiomePaletteMax - 1`](#schematicBiomePaletteMax). While not required, it is highly recommended to include a palette in order to tightly pack the biome id's included in the data array.
+<a name="schematicBiomeData"></a>BiomeData | `varint[]` | Specifies the main storage array which contains `Width * Height * Length` entries for `Biome`s at positions. Each entriy is specified as a varint and refers to an index within [BiomePalette](#schematicBiomePalette). The entries are indexed by `x + z * Width + y * Width * Length`.
```

Comment:
Not really sure it's necessary to specify this in 3D. The reference implementation handles this very oddly atm as well. imho the y-axis should just be dropped until it's reasonably needed in a later version (if vertically distinguished biomes become a thing).